From 3d456d170ab87325a6b4625166384d9fbacb1ca8 Mon Sep 17 00:00:00 2001
From: Michel Thebeau <michel.thebeau@windriver.com>
Date: Wed, 9 Jan 2013 11:29:32 -0500
Subject: [PATCH] imx6/fsl: #ifdef use of platform_pad_change function pointer

On fsl ppc boards (non-imx):
drivers/mmc/host/sdhci-esdhc.h:87:15: error: dereferencing pointer to incomplete type
drivers/mmc/host/sdhci-esdhc.h:88:12: error: dereferencing pointer to incomplete type

Function pointer platform_pad_change is added to struct
esdhc_platform_data in arch/arm/plat-mxc/include/mach/esdhc.h, so it
should only be used by that arch, so ifdef with this:

config MMC_SDHCI_ESDHC_IMX
        tristate "SDHCI support for the Freescale eSDHC/uSDHC i.MX controller"
        depends on ARCH_MXC || ARCH_MX6

Signed-off-by: Michel Thebeau <michel.thebeau@windriver.com>
---
 drivers/mmc/host/sdhci-esdhc.h |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/mmc/host/sdhci-esdhc.h b/drivers/mmc/host/sdhci-esdhc.h
index b534959..e1ce00b 100644
--- a/drivers/mmc/host/sdhci-esdhc.h
+++ b/drivers/mmc/host/sdhci-esdhc.h
@@ -47,9 +47,11 @@ static inline void esdhc_set_clock(struct sdhci_host *host, unsigned int clock)
 	int pre_div = 2;
 	int div = 1;
 	u32 temp;
+#ifdef CONFIG_MMC_SDHCI_ESDHC_IMX
 	struct esdhc_platform_data *boarddata;
 
 	boarddata = host->mmc->parent->platform_data;
+#endif
 
 	if (clock == 0)
 		goto out;
@@ -78,6 +80,7 @@ static inline void esdhc_set_clock(struct sdhci_host *host, unsigned int clock)
 	sdhci_writel(host, temp, ESDHC_SYSTEM_CONTROL);
 	mdelay(1);
 
+#ifdef CONFIG_MMC_SDHCI_ESDHC_IMX
 	/* if there's board callback function
 	 * for pad setting change, that means
 	 * board needs to reconfig its pad for
@@ -85,6 +88,7 @@ static inline void esdhc_set_clock(struct sdhci_host *host, unsigned int clock)
 	 */
 	if (boarddata->platform_pad_change)
 		boarddata->platform_pad_change(clock);
+#endif
 out:
 	host->clock = clock;
 }
-- 
1.7.9.7

